File: sorting/bubble-sort/bubble_sort.jl
#!/usr/local/bin/julia

function bubbleSort(array::Array{Int64, 1})
    array_length = length(array)
    for iteration in 1:array_length
        for item in 1:array_length-iteration-1
            if array[item] > array[item+1]
                array[item], array[item+1] = array[item+1], array[item]
            end
        end
    end
    return array
end 

# julia> a = [12,24,48,11,23,45,9,7,5,78]
# julia> bubbleSort(a)
# 10-element Array{Int64,1}:
#   5
#   7
#   9
#  11
#  12
#  23
#  24
#  45
#  48
#  78

File: sorting/bubble-sort/bubble_sort.py
#!/usr/bin/local/python3

def bubble_sort(array):
    """
    Bubble Sort in Python3!
    """
    array_length = len(array)

    # iterate over the array
    for iteration in range(array_length):
        # for each iteraion, compare each pair of items in this array
        for item in range(0, array_length-iteration-1):
            # if the item is greater, swap it
            if array[item] > array[item+1]:
                array[item], array[item+1] = array[item+1], array[item]
    
    return array

if __name__ == "__main__":
    array = [12,24,48,11,23,45,9,7,5,78]
    print(bubble_sort(array))